Build fails via freeorion-git package for archlinux in aur repository

The build failed with the following error message. Please advise:

In file included from /usr/include/boost/phoenix/object/detail/cpp03/preprocessed/construct.hpp:12:0,
from /usr/include/boost/phoenix/object/detail/cpp03/construct.hpp:6,
from /usr/include/boost/phoenix/object/construct.hpp:76,
from /usr/include/boost/phoenix/object.hpp:12,
from /usr/include/boost/phoenix/phoenix.hpp:15,
from /usr/include/boost/phoenix.hpp:11,
from /usr/include/boost/spirit/include/phoenix.hpp:11,
from /tmp/yaourt-tmp-mancar/aur-freeorion-git/src/freeorion.git/parse/ConditionParserImpl.h:6,
from /tmp/yaourt-tmp-mancar/aur-freeorion-git/src/freeorion.git/parse/ValueRefParserImpl.h:3,
from /tmp/yaourt-tmp-mancar/aur-freeorion-git/src/freeorion.git/parse/DoubleComplexValueRefParser.cpp:1:
/usr/include/boost/phoenix/object/detail/cpp03/preprocessed/construct_10.hpp: In instantiation of ‘const typename boost::phoenix::expression::construct<boost::phoenix::detail::target<T>, A0>::type boost::phoenix::construct(const A0&) [with T = std::__cxx11::basic_string<char>; A0 = char [21]; typename boost::phoenix::expression::construct<boost::phoenix::detail::target<T>, A0>::type = boost::phoenix::actor<boost::proto::exprns_::basic_expr<boost::phoenix::tag::construct, boost::proto::argsns_::list2<boost::proto::exprns_::basic_expr<boost::proto::tagns_::tag::terminal, boost::proto::argsns_::term<boost::phoenix::detail::target<std::__cxx11::basic_string<char> > >, 0l>, boost::proto::exprns_::basic_expr<boost::proto::tagns_::tag::terminal, boost::proto::argsns_::term<char*>, 0l> >, 2l> >]’:
/tmp/yaourt-tmp-mancar/aur-freeorion-git/src/freeorion.git/parse/DoubleComplexValueRefParser.cpp:58:96: required from here
/usr/include/boost/phoenix/object/detail/cpp03/preprocessed/construct_10.hpp:23:25: error: invalid conversion from ‘const char*’ to ‘param_type {aka char*}’ [-fpermissive]
expression::
~~~~~~~~~~~~
construct<detail::target<T>, A0>::
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make(detail::target<T>(), a0);
~~~~^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/boost/phoenix/core/argument.hpp:14:0,
from /usr/include/boost/phoenix/core.hpp:15,
from /usr/include/boost/spirit/include/phoenix_core.hpp:11,
from /usr/include/boost/spirit/home/support/terminal.hpp:17,
from /usr/include/boost/spirit/home/support/common_terminals.hpp:15,
from /usr/include/boost/spirit/home/lex/lexer/terminals.hpp:13,
from /usr/include/boost/spirit/home/lex/lexer.hpp:13,
from /usr/include/boost/spirit/home/lex.hpp:13,
from /usr/include/boost/spirit/home/lex/lexer_lexertl.hpp:15,
from /usr/include/boost/spirit/include/lex_lexertl.hpp:16,
from /tmp/yaourt-tmp-mancar/aur-freeorion-git/src/freeorion.git/parse/Lexer.h:6,
from /tmp/yaourt-tmp-mancar/aur-freeorion-git/src/freeorion.git/parse/ValueRefParser.h:4,
from /tmp/yaourt-tmp-mancar/aur-freeorion-git/src/freeorion.git/parse/ValueRefParserImpl.h:1,
from /tmp/yaourt-tmp-mancar/aur-freeorion-git/src/freeorion.git/parse/DoubleComplexValueRefParser.cpp:1:
/usr/include/boost/phoenix/core/expression.hpp:63:21: note: initializing argument 2 of ‘static boost::phoenix::expr_ext<Actor, Tag, A>::type boost::phoenix::expr_ext<Actor, Tag, A>::make(typename boost::call_traits<A>::param_type ...) [with Actor = boost::phoenix::actor; Tag = boost::phoenix::tag::construct; A = {boost::phoenix::detail::target<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, char*}; boost::phoenix::expr_ext<Actor, Tag, A>::type = boost::phoenix::actor<boost::proto::exprns_::basic_expr<boost::phoenix::tag::construct, boost::proto::argsns_::list2<boost::proto::exprns_::basic_expr<boost::proto::tagns_::tag::terminal, boost::proto::argsns_::term<boost::phoenix::detail::target<std::__cxx11::basic_string<char> > >, 0l>, boost::proto::exprns_::basic_expr<boost::proto::tagns_::tag::terminal, boost::proto::argsns_::term<char*>, 0l> >, 2l> >]’
static type make(typename call_traits<A>::param_type... a)
^~~~
In file included from /usr/include/boost/phoenix/object/detail/cpp03/preprocessed/construct.hpp:12:0,
from /usr/include/boost/phoenix/object/detail/cpp03/construct.hpp:6,
from /usr/include/boost/phoenix/object/construct.hpp:76,
from /usr/include/boost/phoenix/object.hpp:12,
from /usr/include/boost/phoenix/phoenix.hpp:15,
from /usr/include/boost/phoenix.hpp:11,
from /usr/include/boost/spirit/include/phoenix.hpp:11,
from /tmp/yaourt-tmp-mancar/aur-freeorion-git/src/freeorion.git/parse/ConditionParserImpl.h:6,
from /tmp/yaourt-tmp-mancar/aur-freeorion-git/src/freeorion.git/parse/ValueRefParserImpl.h:3,
from /tmp/yaourt-tmp-mancar/aur-freeorion-git/src/freeorion.git/parse/DoubleComplexValueRefParser.cpp:1:
/usr/include/boost/phoenix/object/detail/cpp03/preprocessed/construct_10.hpp: In instantiation of ‘const typename boost::phoenix::expression::construct<boost::phoenix::detail::target<T>, A0>::type boost::phoenix::construct(const A0&) [with T = std::__cxx11::basic_string<char>; A0 = char [22]; typename boost::phoenix::expression::construct<boost::phoenix::detail::target<T>, A0>::type = boost::phoenix::actor<boost::proto::exprns_::basic_expr<boost::phoenix::tag::construct, boost::proto::argsns_::list2<boost::proto::exprns_::basic_expr<boost::proto::tagns_::tag::terminal, boost::proto::argsns_::term<boost::phoenix::detail::target<std::__cxx11::basic_string<char> > >, 0l>, boost::proto::exprns_::basic_expr<boost::proto::tagns_::tag::terminal, boost::proto::argsns_::term<char*>, 0l> >, 2l> >]’:
/tmp/yaourt-tmp-mancar/aur-freeorion-git/src/freeorion.git/parse/DoubleComplexValueRefParser.cpp:67:98: required from here
/usr/include/boost/phoenix/object/detail/cpp03/preprocessed/construct_10.hpp:23:25: error: invalid conversion from ‘const char*’ to ‘param_type {aka char*}’ [-fpermissive]
expression::
~~~~~~~~~~~~
construct<detail::target<T>, A0>::
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make(detail::target<T>(), a0);
~~~~^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/boost/phoenix/core/argument.hpp:14:0,
from /usr/include/boost/phoenix/core.hpp:15,
from /usr/include/boost/spirit/include/phoenix_core.hpp:11,
from /usr/include/boost/spirit/home/support/terminal.hpp:17,
from /usr/include/boost/spirit/home/support/common_terminals.hpp:15,
from /usr/include/boost/spirit/home/lex/lexer/terminals.hpp:13,
from /usr/include/boost/spirit/home/lex/lexer.hpp:13,
from /usr/include/boost/spirit/home/lex.hpp:13,
from /usr/include/boost/spirit/home/lex/lexer_lexertl.hpp:15,
from /usr/include/boost/spirit/include/lex_lexertl.hpp:16,
from /tmp/yaourt-tmp-mancar/aur-freeorion-git/src/freeorion.git/parse/Lexer.h:6,
from /tmp/yaourt-tmp-mancar/aur-freeorion-git/src/freeorion.git/parse/ValueRefParser.h:4,
from /tmp/yaourt-tmp-mancar/aur-freeorion-git/src/freeorion.git/parse/ValueRefParserImpl.h:1,
from /tmp/yaourt-tmp-mancar/aur-freeorion-git/src/freeorion.git/parse/DoubleComplexValueRefParser.cpp:1:
/usr/include/boost/phoenix/core/expression.hpp:63:21: note: initializing argument 2 of ‘static boost::phoenix::expr_ext<Actor, Tag, A>::type boost::phoenix::expr_ext<Actor, Tag, A>::make(typename boost::call_traits<A>::param_type ...) [with Actor = boost::phoenix::actor; Tag = boost::phoenix::tag::construct; A = {boost::phoenix::detail::target<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, char*}; boost::phoenix::expr_ext<Actor, Tag, A>::type = boost::phoenix::actor<boost::proto::exprns_::basic_expr<boost::phoenix::tag::construct, boost::proto::argsns_::list2<boost::proto::exprns_::basic_expr<boost::proto::tagns_::tag::terminal, boost::proto::argsns_::term<boost::phoenix::detail::target<std::__cxx11::basic_string<char> > >, 0l>, boost::proto::exprns_::basic_expr<boost::proto::tagns_::tag::terminal, boost::proto::argsns_::term<char*>, 0l> >, 2l> >]’
static type make(typename call_traits<A>::param_type... a)
^~~~
make[2]: *** [parse/CMakeFiles/freeorionparse.dir/build.make:351: parse/CMakeFiles/freeorionparse.dir/DoubleComplexValueRefParser.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:285: parse/CMakeFiles/freeorionparse.dir/all] Error 2
make: *** [Makefile:150: all] Error 2
==> ERROR: A failure occurred in build().
Aborting...
==> ERROR: Makepkg was unable to build freeorion-git.

--- a/parse/DoubleComplexValueRefParser.cpp
+++ b/parse/DoubleComplexValueRefParser.cpp
@@ -1,6 +1,9 @@
 #include "ValueRefParserImpl.h"
 
 namespace parse {
+    const std::string TOK_SPECIES_EMPIRE_OPINION{"SpeciesEmpireOpinion"};
+    const std::string TOK_SPECIES_SPECIES_OPINION{"SpeciesSpeciesOpinion"};
+
     struct double_complex_parser_rules {
         double_complex_parser_rules() {
             qi::_1_type _1;
@@ -55,7 +58,7 @@ namespace parse {
 
             species_empire_opinion
                 = (
-                    (  tok.SpeciesOpinion_ [ _a = construct<std::string>("SpeciesEmpireOpinion") ]
+                    (  tok.SpeciesOpinion_ [ _a = construct<std::string>(TOK_SPECIES_EMPIRE_OPINION) ]
                        >  parse::label(Species_token) >  string_value_ref [ _d = _1 ]
                     )
                   >> parse::label(Empire_token)  >  simple_int [ _b = _1 ]
@@ -64,7 +67,7 @@ namespace parse {
 
             species_species_opinion
                 = (
-                    (   tok.SpeciesOpinion_ [ _a = construct<std::string>("SpeciesSpeciesOpinion") ]
+                    (   tok.SpeciesOpinion_ [ _a = construct<std::string>(TOK_SPECIES_SPECIES_OPINION) ]
                       >  parse::label(Species_token) >  string_value_ref [ _d = _1 ]
                     )
                   >> parse::label(Species_token) >  string_value_ref [ _e = _1 ]

--- a/parse/StringValueRefParser.cpp
+++ b/parse/StringValueRefParser.cpp
@@ -67,7 +67,10 @@ void initialize_nonnumeric_expression_parsers<std::string>(
         ;
 }
 
-namespace { struct string_parser_rules {
+namespace {
+    const std::string TOK_CURRENT_CONTENT{"CurrentContent"};
+
+    struct string_parser_rules {
         string_parser_rules() {
             qi::_1_type _1;
             qi::_val_type _val;
@@ -93,7 +96,7 @@ namespace { struct string_parser_rules {
                     |   tok.ThisTech_
                     |   tok.ThisSpecies_
                     |   tok.ThisSpecial_
-                   ) [ _val = new_<ValueRef::Constant<std::string> >("CurrentContent") ]
+                   ) [ _val = new_<ValueRef::Constant<std::string> >(TOK_CURRENT_CONTENT) ]
                 ;
 
             free_variable

